lib/mongoid/contextual/mongo.rb

```diff
@@ -420,20 +426,28 @@ def map_reduce(map, reduce)
       #
       # @return [ Array<Object, Array> ] The plucked values.
       def pluck(*fields)
+        # Multiple fields can map to the same field name. For example, plucking
+        # a field and its _translations field map to the same field in the database.
+        # because of this, we need to keep track of the fields inputted.
+        normalized_field_names = []
         normalized_select = fields.inject({}) do |hash, f|
-          hash[klass.database_field_name(f)] = 1
+          db_fn = klass.database_field_name(f)
+          normalized_field_names.push(db_fn)
+
+          if Mongoid.legacy_pluck_distinct
+            hash[db_fn] = 1
```

Comment:
Although `1` does work, it looks a bit too perl-y. I suggest `true` since Ruby has booleans.

Reply:
Done. I think this was originally done this way since it represents a MongoDB projection, which is usually of the form `key: [0,1]`. Both work though.